relay-event-normalization/src/normalize/processor.rs

```diff
@@ -31,24 +33,43 @@ use crate::span::tag_extraction::{self, extract_span_tags};
 use crate::timestamp::TimestampProcessor;
 use crate::utils::MAX_DURATION_MOBILE_MS;
 use crate::{
-    breakdowns, schema, span, transactions, trimming, user_agent, BreakdownsConfig,
-    ClockDriftProcessor, DynamicMeasurementsConfig, GeoIpLookup, RawUserAgentInfo,
+    breakdowns, end_all_spans, normalize_transaction_name, schema, set_default_transaction_source,
+    span, trimming, user_agent, validate_transaction, BreakdownsConfig, ClockDriftProcessor,
+    DynamicMeasurementsConfig, GeoIpLookup, RawUserAgentInfo, TransactionNameConfig,
 };
 
 use crate::LightNormalizationConfig;
 
 /// Configuration for [`NormalizeProcessor`].
 #[derive(Clone, Debug, Default)]
-pub struct NormalizeProcessorConfig<'a> {
+pub(crate) struct NormalizeProcessorConfig<'a> {
     /// Light normalization config.
     // XXX(iker): we should split this config appropriately.
-    light_config: LightNormalizationConfig<'a>,
+    pub light_config: LightNormalizationConfig<'a>,
+
+    /// Configuration to apply to transaction names, especially around sanitizing.
+    pub transaction_name_config: TransactionNameConfig<'a>,
+
+    /// Timestamp range in which a transaction must end.
+    ///
+    /// Transactions that finish outside of this range are considered invalid.
+    /// This check is skipped if no range is provided.
+    pub transaction_range: Option<Range<UnixTimestamp>>,
 }
 
 impl<'a> From<LightNormalizationConfig<'a>> for NormalizeProcessorConfig<'a> {
-    fn from(config: LightNormalizationConfig<'a>) -> Self {
+    fn from(mut config: LightNormalizationConfig<'a>) -> Self {
+        // HACK(iker): workaround to avoid cloning of config items. We'll get
+        // rid of this when we remove light normalization in the next step.
+        let transaction_name_config = config.transaction_name_config;
+        config.transaction_name_config = TransactionNameConfig::default();
+        let transaction_range = config.transaction_range;
+        config.transaction_range = None;
```

Comment:
Something like this would work as well, much shorter and easier to understand I think. 

```suggestion
        let transaction_name_config = std::mem::take(&mut config.transaction_name_config);
        let transaction_range = config.transaction_range.take();
```